py/dml/structure.py

```diff
@@ -1196,8 +1204,9 @@ def default_sym(self, indices):
         ancestry_path = self.trait_node.traits.ancestry_paths[
             default_def_trait][0]
         if default_def_trait is not self.trait_method.vtable_trait:
-            ancestry_path += default_def_trait.ancestry_paths[
-                self.trait_method.vtable_trait][0]
+            # mustn't use +=!
+            ancestry_path = (ancestry_path + default_def_trait.ancestry_paths[
+                self.trait_method.vtable_trait][0])
```

Comment:
I ran into a bug I _thought_ was caused by this, but turned out it wasn't. Nevertheless, this is **dangerous**, so I kept this diff.

Reply:
not actually a bug. Will be replaced with a comment explaining the use of `+=` is safe.